contracts/MyToken_721.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+contract MyToken_721 is ERC721Enumerable, Ownable, ReentrancyGuard {
+
+    uint private _tokenPrice;
+    string private _tokenBaseUri;
+    mapping(address => uint) private _minters; 
+
+    constructor(string memory baseUri, uint tokenPrice) ERC721("MyToken", "MTK") {
+        _tokenBaseUri = baseUri;
+        _tokenPrice = tokenPrice;
+    }
+
+    function safeMint() public payable nonReentrant {
+        uint mintCount = _minters[msg.sender];

```

Comment:
just a minor comment - no need to have a tmp `mintCount` variable, just use `_minters[msg.sender]` in `if` statement.